keras_cv/models/resnet.py

```diff
@@ -0,0 +1,537 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ResNet models for Keras.
+Reference:
+  - [Deep Residual Learning for Image Recognition](https://arxiv.org/abs/1512.03385) (CVPR 2015)
+  - [Identity Mappings in Deep Residual Networks](https://arxiv.org/abs/1603.05027) (ECCV 2016)
+  - [Based on the original keras.applications ResNet](https://github.com/keras-team/keras/blob/master/keras/applications/py)
+"""
+
+import tensorflow as tf
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+BN_AXIS = 3
+
+BASE_DOCSTRING = """Instantiates the {name} architecture.
+    Reference:
+        - [Deep Residual Learning for Image Recognition](https://arxiv.org/abs/1512.03385)
+        - [Identity Mappings in Deep Residual Networks](https://arxiv.org/abs/1603.05027) (ECCV 2016)
+    This function returns a Keras {name} model.
+    For transfer learning use cases, make sure to read the [guide to transfer
+        learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+    Args:
+        include_rescaling: whether or not to Rescale the inputs.If set to True,
+            inputs will be passed through a `Rescaling(1/255.0)` layer.
+        include_top: whether to include the fully-connected layer at the top of the
+            network.  If provided, num_classes must be provided.
+        num_classes: optional number of classes to classify images into, only to be
+            specified if `include_top` is True, and if no `weights` argument is
+            specified.
+        weights: one of `None` (random initialization), or a pretrained weight file
+            path.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        pooling: optional pooling mode for feature extraction
+            when `include_top` is `False`.
+            - `None` means that the output of the model will be the 4D tensor output
+                of the last convolutional block.
+            - `avg` means that global average pooling will be applied to the output
+                of the last convolutional block, and thus the output of the model will
+                be a 2D tensor.
+            - `max` means that global max pooling will be applied.
+        name: (Optional) name to pass to the model.  Defaults to "{name}".
+    Returns:
+      A `keras.Model` instance.
+"""
+
+
+def ResNet(
+    stack_fn,
+    preact,
+    include_rescaling,
+    name="ResNet",
+    include_top=True,
+    weights="None",
+    input_shape=(None, None, 3),
+    pooling=None,
+    num_classes=1000,
+    classifier_activation="softmax",
+    **kwargs,
+):
+    """Instantiates the ResNet, ResNetV2, and ResNeXt architecture.
```

Comment:
Do we have an example of ResNeXt?

Reply:
If we have It will cover https://github.com/keras-team/keras-cv/issues/37

Reply:
I removed the Resnext block from the file, I'll change the docstring as well. I thought it would be best that we support ResNetv1&2  for the time being, as ResneXt is not supported by `keras.applications` as of now either.

Reply:
great, yeah update the docstring please!